src/main/java/Data/UserManager.java

```diff
@@ -0,0 +1,61 @@
+package Data;
+
+import Library.Book;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class UserManager {
+
+    private final String filePath = "C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\users.txt";
+
+    private List<User> userList;
+
+    public UserManager() {
+        this.userList = readAllUsers();
+    }
+
+    public User registerUser(String username, String password, String email) {
+        String userID = UUID.randomUUID().toString();
+        User newUser = new User(userID, username, password, email);
+
+        try {
+            FileWriter writer = new FileWriter("C:\\Users\\david\\IdeaProjects\\LibraryProject\\Online-Library\\src\\main\\resources\\users.txt", true);
+            writer.write(newUser + "\n");
+            writer.close();
+        } catch (IOException exception) {
+            exception.printStackTrace();
```

Comment:
Not user friendly. Look for another exception handling

Reply:
New Exception Class created: 

package repositories;

import java.io.IOException;

public class InvalidNewUserException extends RuntimeException{
    public InvalidNewUserException (String message, IOException cause){
        super(message,cause);
    }
}

Block of code changed to: 
    @Override
    public void saveBookInfo(Book book) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(filePath, true))) {
            String formattedEntry = String.format("\"%s\" %s - %s", book.getTitle(), book.getAuthor(), book.getBookStatus());
            writer.println(formattedEntry);
        } catch (IOException exception) {
            throw new BookSaveException ("An error occurred while saving the book information.", exception);
        }
    }

Reply:
Since BookSaveException and InvalidNewUserException have the same logic I refactored one of them to NewItemSaveException and used it in both cases.